vpr/src/base/write_interchange_netlist.cpp

```diff
@@ -0,0 +1,532 @@
+        if (modelPort->is_clock) {
+            port.properties["is_clock"] = "1";
+        }
+        if (modelPort->is_non_clock_global) {
+            port.properties["is_non_clock_global"] = "1";
+        }
+
+        // Add the port
+        port.index = ports_.size();
+        ports_.emplace(std::make_pair(std::make_pair(modelName, modelPort->name), port));
+    }
+
+    uint32_t addCellInstance(const AtomBlockId& block) {
+        const t_model* model = netlist_.block_model(block);
+        VTR_ASSERT(model != nullptr);
+
+        // Create the cell instance
+        CellInstance inst;
+        inst.name = netlist_.block_name(block);
+        inst.type = model->name;
+
+        // Collect attributes and parameters
+        auto addProperty = [&](const std::pair<std::string, std::string>& prop) {
+            if (inst.properties.count(prop.first)) {
+                VTR_LOG_WARN("Duplicate property '%s'='%s' of cell instance '%s'\n",
+                             prop.first.c_str(), prop.second.c_str(), inst.name.c_str());
+            } else {
+                inst.properties.insert(prop);
+            }
+        };
+
+        // FIXME: The FPGA Interchange logical netlist format does not
+        // differentiate between attributes and parameters.
+        for (const auto& it : netlist_.block_attrs(block)) {
+            addProperty(it);
+        }
+        for (const auto& it : netlist_.block_params(block)) {
+            addProperty(it);
+        }
+
+        // Add the cell instance
+        inst.index = cellInstances_.size();
+        VTR_ASSERT(cellInstances_.count(inst.name) == 0);
+        cellInstances_.emplace(std::make_pair(inst.name, inst));
+
+        return inst.index;
+    }
+
+    void addNet(const AtomNetId& net_id) {
+        Net net;
+
+        // Handle driver / endpoints
+        for (const auto& pin_id : netlist_.net_pins(net_id)) {
+            // Block
+            const auto& blk_id = netlist_.pin_block(pin_id);
+            const auto& blk_name = netlist_.block_name(blk_id);
+
+            // Port / bit
+            const auto& port_id = netlist_.pin_port(pin_id);
+            const auto& port_bit = netlist_.pin_port_bit(pin_id);
+            const auto& port_name = netlist_.port_name(port_id);
+
+            // Find the cell declaration
+            const t_model* model = netlist_.block_model(blk_id);
+            VTR_ASSERT(model != nullptr);
+
+            PortInstance portInst;
+            portInst.cell = blk_name;
+            portInst.port = port_name;
+            portInst.bit = port_bit;
+            portInst.isTop = isIoModel(model->name);
+
+            net.ports[model->name].push_back(portInst);
+        }
+
+        // Add the net
+        const auto& name = netlist_.net_name(net_id);
+        VTR_ASSERT(nets_.count(name) == 0);
+        nets_.emplace(std::make_pair(name, net));
+    }
+
+    uint32_t buildTopLevelCellDeclaration() {
```

Comment:
Where do you use return value of this function?
Is it something that is left from previous iterations?